astropy/time/formats.py

```diff
@@ -738,18 +738,38 @@ def __new__(cls, val1, val2, scale, precision,
 
         if scale is None:
             scale = val1_0.scale
+
         if val1.shape:
             vals = [getattr(val, scale)._time for val in val1]
             jd1 = np.concatenate([np.atleast_1d(val.jd1) for val in vals])
             jd2 = np.concatenate([np.atleast_1d(val.jd2) for val in vals])
+
+            # Collect individual location values and merge into a single location.
+            if any(tm.location is not None for tm in val1):
+                if any(tm.location is None for tm in val1):
+                    raise ValueError('cannot concatenate times unless all locations '
+                                     'are set or no locations are set')
+                locations = []
+                for tm in val1:
+                    location = np.broadcast_to(tm.location, tm._time.jd1.shape,
+                                               subok=True)
+                    locations.append(np.atleast_1d(location))
+                location = np.concatenate(locations)
```

Comment:
This now works for numpy 1.17 but not yet for numpy 1.16 without explicit opt-in to numpy function overrides. The easiest way to solve it should be to just use `location = u.Quantity(locations)` instead (or, inside an `if NUMPY_LT_1_17` branch).

Reply:
Hmm, my solution does not actually work...  I'll push up a more detailed fix in a bit.